app/gui2/shared/yjsModel.ts

```diff
@@ -168,50 +170,51 @@ export class DistributedModule {
 }
 
 export type SourceRange = readonly [start: number, end: number]
+declare const brandSourceRangeKey: unique symbol
+export type SourceRangeKey = string & { [brandSourceRangeKey]: never }
+
+export function sourceRangeKey(range: SourceRange): SourceRangeKey {
+  return `${range[0].toString(16)}:${range[1].toString(16)}` as SourceRangeKey
+}
+export function sourceRangeFromKey(key: SourceRangeKey): SourceRange {
+  return key.split(':').map((x) => parseInt(x, 16)) as [number, number]
+}
```

Comment:
Why do we need branding here? May we mismatch it with another pair of numbers? Perhaps we could just use object with `start` and `end` property (as it was once in CB code)?

Reply:
SourceRange is just a pair of numbers. SourceRangeKey is a branded string containing a serialized SourceRange for use as map keys etc. Maybe we would never mix it up with strings not produced by `sourceRangeKey`, but it doesn't hurt to give it a type.

Reply:
Ah, I misread that, indeed this is brand for string. Never mind then.